elastic/kibana
# [ResponseOps] [Alerting] Add warning to Edit Rule Flyout when publicUrl is not configured

Resolves https://github.com/elastic/kibana/issues/144890

## Summary

Added a warning to the rule flyout that will warn users about using `context.alertDetailsUrl` and `context.viewInAppUrl` in an input or text area when they do not have `publicUrl` set in their kibana config.

**Warning:**
<img width="605" alt="Screen Shot 2023-02-01 at 11 18 17 AM" src="https://user-images.githubusercontent.com/109488926/216102190-f41b4af0-5e40-4529-8668-68e2f57c5046.png">

<img width="586" alt="Screen Shot 2023-02-01 at 11 18 26 AM" src="https://user-images.githubusercontent.com/109488926/217331719-2f5b18e3-62e8-42be-b389-101538e05b9b.png">

### Checklist

Delete any items that are not applicable to this PR.



### To verify

- Remove your kibana config for `server.publicBaseUrl`, and create a rule with a connector. Add the context variables `context.alertDetailsUrl` or `context.viewInAppUrl` and verify that a warning shows up in the form.
- Add `server.publicBaseUrl` to your kibana config, and create a rule with a connector. Add the context variables `context.alertDetailsUrl` or `context.viewInAppUrl` and verify that a warning shows does **not** show in the form.
- You should try this for multiple connectors to verify that the warnings are configured correctly for them.
